File: lint-arwaky/archive/taxonomy/source_analysis_vo.py
from pydantic import BaseModel, ConfigDict, Field
from .lint_position_vo import LineNumber, ColumnNumber
from .error_value_vo import ModuleName, PrimitiveTypeName
from .symbol_name_vo import SymbolName


class ImportInfo(BaseModel):
    """Information about a single import statement."""

    model_config = ConfigDict(frozen=True)
    line: LineNumber
    module: ModuleName
    name: SymbolName | None = None


class PrimitiveViolation(BaseModel):
    """Information about a forbidden primitive type usage."""

    model_config = ConfigDict(frozen=True)
    line: LineNumber
    column: ColumnNumber
    type_name: PrimitiveTypeName


class ImportInfoList(BaseModel):
    """List of ImportInfo objects."""

    model_config = ConfigDict(frozen=True)
    values: list[ImportInfo] = Field(default_factory=list)

    def __iter__(self):
        return iter(self.values)

    def __len__(self):
        return len(self.values)

    def append(self, item: ImportInfo) -> None:
        """Append an item to the list."""
        self.values.append(item)


class PrimitiveViolationList(BaseModel):
    """List of PrimitiveViolation objects."""

    model_config = ConfigDict(frozen=True)
    values: list[PrimitiveViolation] = Field(default_factory=list)

    def __iter__(self):
        return iter(self.values)

    def __len__(self):
        return len(self.values)

    def append(self, item: PrimitiveViolation) -> None:
        """Append an item to the list."""
        self.values.append(item)

File: lint-arwaky/src-rust/taxonomy/source_analysis_vo.rs
use serde::{Serialize, Deserialize};
use super::*;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ImportInfo {
    pub line: LineNumber,
    pub module: String,
    #[serde(default)]
    pub name: Option<String>,
}

impl ImportInfo {
    pub fn new(line: LineNumber, module: String) -> Self {
        Self { line, module, name: None }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct PrimitiveViolation {
    pub line: LineNumber,
    pub column: ColumnNumber,
    pub type_name: String,
}

impl PrimitiveViolation {
    pub fn new(line: LineNumber, column: ColumnNumber, type_name: String) -> Self {
        Self { line, column, type_name }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ImportInfoList {
    #[serde(default)]
    pub values: Vec<ImportInfo>,
}

impl ImportInfoList {
    pub fn new() -> Self { Self { values: Vec::new() } }
    pub fn push(&mut self, item: ImportInfo) { self.values.push(item); }
    pub fn len(&self) -> usize { self.values.len() }
    pub fn is_empty(&self) -> bool { self.values.is_empty() }
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct PrimitiveViolationList {
    #[serde(default)]
    pub values: Vec<PrimitiveViolation>,
}

impl PrimitiveViolationList {
    pub fn new() -> Self { Self { values: Vec::new() } }
    pub fn push(&mut self, item: PrimitiveViolation) { self.values.push(item); }
    pub fn len(&self) -> usize { self.values.len() }
    pub fn is_empty(&self) -> bool { self.values.is_empty() }
}
